doc/source/data/getting-started.rst

```diff
@@ -1,229 +1,86 @@
-    import dask.dataframe as dd
-
-    # Create a Dataset from a list of Pandas DataFrame objects.
-    pdf = pd.DataFrame({"one": [1, 2, 3], "two": ["a", "b", "c"]})
-    ds = ray.data.from_pandas([pdf])
-
-    # Create a Dataset from a Dask-on-Ray DataFrame.
-    dask_df = dd.from_pandas(pdf, npartitions=10)
-    ds = ray.data.from_dask(dask_df)
-
-Saving Datasets
-===============
-
-Datasets can be written to local or remote storage using ``.write_csv()``, ``.write_json()``, and ``.write_parquet()``.
-
-.. code-block:: python
+.. literalinclude:: ./doc_code/quick_start.py
+   :language: python
+   :start-after: __create_from_files_begin__`
+   :end-before: __create_from_files_end__
 
-    # Write to csv files in /tmp/output.
-    ray.data.range(10000).write_csv("/tmp/output")
-    # -> /tmp/output/data0.csv, /tmp/output/data1.csv, ...
+Once you have a Dataset, you can save it to local or remote storage in desired format, using ``.write_csv()``, ``.write_json()``, and ``.write_parquet()``.
 
-    # Use repartition to control the number of output files:
-    ray.data.range(10000).repartition(1).write_csv("/tmp/output2")
-    # -> /tmp/output2/data0.csv
+.. literalinclude:: ./doc_code/quick_start.py
+   :language: python
+   :start-after: __save_dataset_begin__
+   :end-before: __save_dataset_end__
 
-You can also convert a ``Dataset`` to Ray-compatible distributed DataFrames:
-
-.. code-block:: python
-
-    # Convert a Ray Dataset into a Dask-on-Ray DataFrame.
-    dask_df = ds.to_dask()
 
 Transforming Datasets
-=====================
-
-Datasets can be transformed in parallel using ``.map_batches()``. Ray will transform
-batches of records in the Dataset using the given function. The function must return
-a batch of records. You are allowed to filter or add additional records to the batch,
-which will change the size of the Dataset.
-
-Transformations are executed *eagerly* and block until the operation is finished.
-
-.. code-block:: python
-
-    def transform_batch(df: pandas.DataFrame) -> pd.DataFrame:
-        return df.applymap(lambda x: x * 2)
+---------------------
 
-    ds = ray.data.range_arrow(10000)
-    ds = ds.map_batches(transform_batch, batch_format="pandas")
-    # -> Map Progress: 100%|████████████████████| 200/200 [00:00<00:00, 1927.62it/s]
-    ds.take(5)
-    # -> [{'value': 0}, {'value': 2}, ...]
+Once you have a Dataset, you can transform it by applying a user-defined function and produce another Dataset.
+Under the hood the transformation is executed in parrallel for performance at scale.
 
-The batch format can be specified using ``batch_format`` option, which defaults to "native",
-meaning pandas format for Arrow-compatible batches, and Python lists for other types. You
-can also specify explicitly "arrow" or "pandas" to force a conversion to that batch format.
-The batch size can also be chosen. If not given, the batch size will default to entire blocks.
+.. literalinclude:: ./doc_code/quick_start.py
```

Comment:
Could we make the example code not horizontal scroll? Consider trimming the progress bar / line wrap the show() output.

Reply:
Yep, code reformatted and black fmt off for the code block.